feat(replay): Update docs for new Replay APIs (#6735)
getsentry/sentry-docs: src/platforms/javascript/common/session-replay/understanding-sessions.mdx

---
title: Understanding Sessions
sidebar_order: 4300
notSupported:
  - javascript.cordova
  - javascript.wasm
description: 'Learn about customizing sessions with the Session Replay SDK.'
---

> In most cases, setting the Replay sample rates will be all you need to do in order to capture recording data.

Reviewer: ```suggestion
In most cases, setting the Replay sample rates will be all you need to do in order to capture recording data you need.
```

Author: hmm, that's `you need` twice 🤔

Author: what about: 
```suggestion
In most cases, setting the Replay sample rates will be all you need to do in order to capture recording data you care about.
```

?

Reviewer: Sounds good!

Reviewer 2: We can shorten it even more: 
In most cases, setting the Replay sample rates will be all you need to do to begin capturing recording data you care about.

Author: ```suggestion
In most cases, setting the Replay sample rates will be all you need to do to begin capturing recording data you care about.
```

If both `replaysSessionSampleRate` and `replaysOnErrorSampleRate` are `0`, the recording won't start, but you'll be able to <PlatformLink to="/session-replay/understanding-sessions/#manually-starting-recording">record manually</PlatformLink>.

## Session Mode

In `session` mode, Replay will continuously record and send data to Sentry. After 15 minutes of user inactivity, or a maximum duration of 60 minutes, the session will end and a new session will be initialized based on the rules from the section above.

## Buffer Mode

In `buffer` mode, Replay will continuously record data, but won't send any of it. It will instead keep up to the last 60 seconds in memory. When an error occurs, `replaysOnErrorSampleRate` will be checked and if it's sampled, the replay will be uploaded to Sentry and continue recording normally. After 15 minutes of user inactivity, or a maximum duration of 60 minutes, the session will end and the replay will stop.

## Manually Starting Replay

> If both `replaysSessionSampleRate` and `replaysOnErrorSampleRate` are `0`, the recording won't start, but you'll be able to record manually:

Reviewer: You can manually start a replay even if your Replay sample rates are greater than 0, right? If so, I wouldn't use this as the text here. We already mention this point above under "Default Session Initialization" :)

Author: So I changed this to:

> You can manually start a Replay session with:

plus below the code snippet:

> This can be used either if both `replaysSessionSampleRate` and `replaysOnErrorSampleRate` are `0` and thus no session has been started automatically, or if you previously stopped a session and want to start a new one (see below). `start` and `startBuffering` will throw an error if a session is currently running.

---
title: Understanding Sessions
sidebar_order: 4300
notSupported:
  - javascript.cordova

>   - javascript.electron

Reviewer 3: We support Electron:

```suggestion
```

Author: Ah, need to remove this from the other docs as well then. will do!

  - javascript.wasm
description: 'Learn about customizing sessions with the Session Replay SDK.'
---

## Default Session Initialization

By default, Replay will immediately start a session in either `session` or `buffer` mode, depending on your `replaysSessionSampleRate` and `replaysOnErrorSampleRate`.

When Replay is initialized, we check the `replaysSessionSampleRate`.
If it is sampled, then we start recording in `session` mode & sending Replay data immediately.

> Else, if `replaysOnErrorSampleRate > 0`, we'll start recording in `buffer` mode.

Reviewer 4: ```suggestion
Else, if `replaysOnErrorSampleRate > 0`, it will start recording in `buffer` mode.
```

> In this mode, whenever an error occurs we'll check `replaysOnErrorSampleRate`.
> If it is sampled, when we'll upload the Replay to Sentry and continue recording normally.

Reviewer 4: ```suggestion
In this mode, when an error occurs, `replaysOnErrorSampleRate` is used to check if it should be sampled. If it is sampled, then the Replay will be uploaded to Sentry and continue recording normally.
```

When both `replaysSessionSampleRate` and `replaysOnErrorSampleRate` are `0`, recording will _not_ start.
In this case, you can manually start recording.

## Session Mode

In `session` mode, Replay will continuously record & send data to Sentry. After 15 minutes of user inactivity, or a maximum duration of 60 minutes, the session will end and a new session will be initialized based on the rules from the section above.

## Buffer Mode

> In `buffer` mode, Replay will continuously record data, but only keep up to the last 60 seconds of data in memory. When an error occurs, Replay will check `replaysOnErrorSampleRate` and if it is sampled, Replay will upload the data to Sentry and continue recording normally. After 15 minutes of user inactivity, or a maximum duration of 60 minutes, the session will end and Replay will stop.

Reviewer 4: ```suggestion
In `buffer` mode, Replay will continuously record data, but only keep up to the last 60 seconds of data in memory. When an error occurs, `replaysOnErrorSampleRate` is checked and if it is sampled, the replay will be uploaded to Sentry and continue recording normally. After 15 minutes of user inactivity, or a maximum duration of 60 minutes, the session will end and the replay will stop.
```

> In `buffer` mode, Replay will continuously record data, but will only keep up to the last 60 seconds in memory. When an error occurs, `replaysOnErrorSampleRate` will be checked and if it's sampled, the replay will be uploaded to Sentry and continue recording normally. After 15 minutes of user inactivity, or a maximum duration of 60 minutes, the session will end and the replay will stop.

Reviewer 3: ```suggestion
In `buffer` mode, Replay will continuously record data, but won't send any of it. It will instead keep up to the last 60 seconds in memory. When an error occurs, `replaysOnErrorSampleRate` will be checked and if it's sampled, the replay will be uploaded to Sentry and continue recording normally. After 15 minutes of user inactivity, or a maximum duration of 60 minutes, the session will end and the replay will stop.
```

## Manually Starting Recording

If both `replaysSessionSampleRate` and `replaysOnErrorSampleRate` are `0`, recording will _not_ start. In this case, you can manually start recording:

```javascript
// This initializes Replay without starting any session
const replay = new Sentry.Replay({
  replaysSessionSampleRate: 0,
  replaysOnErrorSampleRate: 0,
});

// This starts in `session` mode, regardless of sample rates
replay.start();

> 

Reviewer 4: ```suggestion

// OR

```

// This starts in `buffer` mode, regardless of sample rates
replay.startBuffering();
```

## Manually Stopping Recording

You can always stop a running session:

```javascript
await replay.stop();
```

> This will flush any pending recording data, and then stop the replay.

Reviewer 4: Should we mention the session?

```suggestion
This will flush any pending recording data, stop the replay, and end the session.
```

---
title: Understanding Sessions
sidebar_order: 4300
notSupported:
  - javascript.cordova
  - javascript.wasm
description: 'Learn about customizing sessions with the Session Replay SDK.'
---

In most cases, setting the Replay sample rates will be all you need to do in order to capture recording data.

> For more complex cases, it is helpful to understand how sessions work and how to manually control them.

Reviewer 2: ```suggestion
For more complex cases, it's helpful to understand how sessions work and how to manually control them.
```

## Default Session Initialization

By default, Replay will immediately start a session in either `session` or `buffer` mode, depending on your `replaysSessionSampleRate` and `replaysOnErrorSampleRate`.

> When Replay is initialized, we check the `replaysSessionSampleRate`. If it's sampled, we start recording and sending Replay data immediately.
> 

Reviewer 2: ```suggestion
When Replay is initialized, we check the `replaysSessionSampleRate`. If it's sampled, we'll start recording and sending Replay data immediately.

```

In this mode, when an error occurs, `replaysOnErrorSampleRate` is used to check if it should be sampled. If it is sampled, then the Replay will be uploaded to Sentry and continue recording normally.

When both `replaysSessionSampleRate` and `replaysOnErrorSampleRate` are `0`, recording will _not_ start.

> In this case, you can manually start recording.

Author: Yeah, I think you're right! I updated the section in "Configuration" to link here instead.

In `session` mode, Replay will continuously record and send data to Sentry. After 15 minutes of user inactivity, or a maximum duration of 60 minutes, the session will end and a new session will be initialized based on the rules from the section above.

## Buffer Mode

In `buffer` mode, Replay will continuously record data, but only keep up to the last 60 seconds of data in memory. When an error occurs, `replaysOnErrorSampleRate` is checked and if it is sampled, the replay will be uploaded to Sentry and continue recording normally. After 15 minutes of user inactivity, or a maximum duration of 60 minutes, the session will end and the replay will stop.

## Manually Starting Recording

If both `replaysSessionSampleRate` and `replaysOnErrorSampleRate` are `0`, recording will _not_ start. In this case, you can manually start recording:

```javascript
// This initializes Replay without starting any session
const replay = new Sentry.Replay({
  replaysSessionSampleRate: 0,
  replaysOnErrorSampleRate: 0,
});

// This starts in `session` mode, regardless of sample rates
replay.start();

// OR

// This starts in `buffer` mode, regardless of sample rates
replay.startBuffering();
```

> ## Manually Stopping Recording

Author: I went with `Manually Stopping Replay`, wdyt?

> ## Manually Stopping Recording

Reviewer 2: Yes! That works.

> You can always stop a running session:

Reviewer 2: ```suggestion
You can always stop a running session with:
```

```javascript
await replay.stop();
```

This will flush any pending recording data, stop the replay, and end the session.

## Manually Flushing Recording Data

> You can flush any pending recording data:

Reviewer 2: ```suggestion
You can flush any pending recording data with:
```

```javascript
await replay.flush();
```

> When in `session` mode, this will upload any pending recording data to Sentry.
> When in `buffer` mode, this will upload any pending recording data to Sentry and then continue recording, the same as if an error is sampled due to `replaysOnErrorSampleRate`.

Reviewer 2: ```suggestion
In `session` mode, this will upload any pending recording data to Sentry. In `buffer` mode, this will upload any pending recording data to Sentry and then continue recording, the same as when an error is sampled with `replaysOnErrorSampleRate`.
```
